.github/workflows/.test.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: test
+on:
+  workflow_call:
+
+jobs:
+  check-style:
+    runs-on: ubuntu-latest
+    container:
+      image: golang:1.20.0-bullseye
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - name: ci/cache-gopath
+      uses: actions/cache@627f0f41f6904a5b1efbaed9f96d9eb58e92e920 # v3.2.4
+      with:
+        key: go-mod-v1-${{ hashFiles('go.sum') }}
+        path: /go
+      if: "${{ !env.ACT }}"
```

Comment:
1/5 that it's easier to drop the specific image from L10 and use `actions/setup-go` instead.

Reply:
This comment seems unresolved, right?

Reply:
@hanzei  Yes sorry I missed this, but could you elaborate more on this because I am not sure I fully follow?

Reply:
Surely. https://github.com/actions/setup-go does provide a workflow that sets up go and does module catching. It's the standard way to setup Go. We already use it for plugins: https://github.com/mattermost/actions/blob/b0a566cc5a6963a6ec931f06ea0bceddef55ebc0/plugin-ci/setup/action.yaml#L55-L59